Providers/Modules/Plugins/Common/plugin/out_oms_blob.rb

```diff
@@ -338,7 +339,15 @@ def handle_records(tag, records)
       @log.debug "Success getting the BLOB information in #{time.round(3)}s"
 
       start = Time.now
-      dataSize, etag = append_blob(blob_uri, records, filePath, blocks_committed)
+
+      # get a lock for the blob append to avoid storage errors when parallel threads are writing
+      BlockLock.lock
```

Comment:
can we lock only when we know that number of threads configured in output plugin are > 1. So in case those are configured we will just upload data right away

Reply:
I have added the check now